go/tools/builders/pack.go

```diff
@@ -162,6 +156,19 @@ func extractFiles(archive, dir string, names map[string]struct{}) (files []strin
 	}
 }
 
+func openArchive(archive string) (*bufio.Reader, io.Closer, error) {
```

Comment:
I think the signature here should be `func openArchive(archive stirng) (io.ReadCloser, error)`.

This function is only used in this file, so perhaps this is overengineering, but I think `readFileInArchive` should return `io.ReadCloser`, and this would make that easier to implement.

A type like this would help combine the `*bufio.Reader` and `*os.File`.

```
type readerWithCloser struct {
    io.Reader
    c io.Closer
}

func (rc *readerWithCloser) Close() error { return rc.c.Close() }
```